### PR TITLE
Add support for migrating Microsoft.AspNetCore.Mvc.ViewCompilation

### PR DESCRIPTION
**Pull request by pranavkm**
Fixes https://github.com/aspnet/MvcPrecompilation/issues/55

### CODE REVIEW

**Comment by pranavkm** [comment]
cc @livarcocc \ @Eilon

File: src/Microsoft.DotNet.ProjectJsonMigration/Rules/MigrateScriptsRule.cs [hunk 1]
```diff
@@ -94,6 +99,16 @@ private bool CommandIsNotNeededInMSBuild(string command)
...
         }
 
+        private static bool IsRazorPrecompilationCommand(string command)
+        {
+            return command.Contains("dotnet razor-precompile");
+        }
+
+        private static void EnableRazorCompilationOnPublish(ProjectRootElement csproj)
+        {
+            csproj.AddProperty("MvcRazorCompileOnPublish", "true");
```
Comment (pranavkm) [inline src/Microsoft.DotNet.ProjectJsonMigration/Rules/MigrateScriptsRule.cs]: This ends up creating a separate `PropertyGroup` at the top of the csproj with the property. Do we care about trying to make it look pretty (and try and merge it in to the first `PropertyGroup`)?

Comment (piotrpMSFT) [inline src/Microsoft.DotNet.ProjectJsonMigration/Rules/MigrateScriptsRule.cs]: We generally care about it looking pretty. Is there a more correct property group than the `first` one?

Comment (livarcocc) [inline src/Microsoft.DotNet.ProjectJsonMigration/Rules/MigrateScriptsRule.cs]: There is a CommonPropertyGroup property at MigrationRuleInputs. Just add this property there.

**Comment by livarcocc** [comment]
Yeah, we need tests for this. Also, shouldn't you also remove a PostPublish step?

**Comment by GuardRex** [comment]
Just curious, the `Microsoft.AspNetCore.Mvc.Razor.ViewCompilation` package isn't available on a public feed yet, correct? I don't see it on `aspnetcore-dev` or any of the CLI repo feeds in use.

**Comment by piotrpMSFT** [comment]
@MattGertz tagging for shiproom approval

**Comment by piotrpMSFT** [comment]
@pranavkm are tests coming soon?

**Comment by piotrpMSFT** [comment]
@dotnet-bot test RHEL7.2 x64 Release Build please.

**Comment by MattGertz** [comment]
I agree.

**Comment by pranavkm** [comment]
@piotrpMSFT updated. Still in the process of running tests locally since previous runs failed with some unrelated errors about .tam files being changed.

**Comment by piotrpMSFT** [comment]
@MattGertz proactive?

**Comment by piotrpMSFT** [comment]
@pranavkm can you fix merge conflicts? Once done, we're ready to merge.

**Comment by pranavkm** [comment]
@piotrpMSFT  - @livarcocc sorted the merged conflicts for me. I guess we can merge it in once the build passes.

**Comment by pranavkm** [comment]
Thanks!

**Pull request merged**